src/api/wallet.js

```diff
@@ -34,6 +34,35 @@ const walletApi = {
     });
   },
 
+  /**
+   * Call API to get address history
+   *
+   * XXX Our current method to allow retry a request demands that we create an axios
+   * instance with a resolve callback, which will be used in case of failure and the
+   * user decides to retry. Because of that, it's impossible to use the old method (getAddressHistory)
+   * to get data with async/await, only with promises. Because of the pagination,
+   * we are in a looping getting data while not finished, so the code with async/await is
+   * much cleaner.
+   *
+   * So, right now to use async/await we must use this method and it's not possible to
```

Comment:
But would the user be able to retry the whole operation? What happens if a problem occur during the fetch?

Reply:
The user would need to reset the wallet and start again. Our retry process depends on the promise resolve method, which does not exist on the async/await call. My opinion is that we should move forward with this and create an issue to handle it.

Reply:
You mean "reset the wallet" or "restart the wallet"? Either way, I don't think we should move forward in this case. It seems an awful user experience.

Reply:
The user is able to close the wallet and open again to try, or click on the button "Reset Wallet" when the error modal opens.

I agree it's bad but I don't think we should do something in this PR. It would required a full rebuild in the retry request method.

We can discuss that we will only release a new wallet version once this is solved, or not. But not in this PR.  And my opinion is that right now this PR would help lot of people. We could release a new version and work on solving this asap.

Reply:
I agree. Let's open an issue to solve this retry issue.

Reply:
Done